sdk/communication/AzureCommunicationChat/README.md

```diff
@@ -1,11 +1,538 @@
+    )]
+)
+client.add(chatThreadMembers: threadMembers, chatThreadId: threadId) { result, _ in
+    switch result {
+    case .success:
+        print("<- SUCCESS! in addThreadMembers.")
+
+    case let .failure(error):
+        print("!! FAIL! in addThreadMembers.")
+        fatalError(error.message)
+    }
+}
+```
+
+### Remove a thread member
+
+Use the `removeChatThreadMember` method to remove a member from a thread.
+
+- `chatThreadId` is the unique ID of the thread.
+- `chatMemberId` is the user ID in the chat thread's member list.
+
+```swift
+client.removeChatThreadMember(chatThreadId: threadId, chatMemberId: memberId) { result, _ in
+    switch result {
+    case .success:
+        print("<- SUCCESS! in removeThreadMember.")
+
+    case let .failure(error):
+        print("!! FAIL! in removeThreadMember.")
+        fatalError(error.message)
+    }
+}
+```
+
+## Events Operations
+
+### Send a typing notification
+
+Use the `sendTypingNotification` method to post a typing notification event to a thread, on behalf of a user.
+
+```swift
+client.sendTypingNotification(chatThreadId: threadId) { result, _ in
+    switch result {
+    case .success:
+        print("<- SUCCESS! in notifyUserTyping.")
+
+    case let .failure(error):
+        print("!! FAIL! in notifyUserTyping.")
+        fatalError(error.message)
+    }
+}
+```
+
+### Send read receipt
+
+Use the `send` method to post a read receipt event to a thread, on behalf of a user.
+
+- `SendReadReceiptRequest` is the model to be passed to this method.
+- `chatMessageId` is the unique ID of the message.
+- `chatThreadId` is the unique ID of the thread.
+
+```swift
+let readReceipt = SendReadReceiptRequest(chatMessageId: messageId)
+
+client.send(chatReadReceipt: readReceipt, chatThreadId: threadId) { result, _ in
+    switch result {
+    case .success:
+        print("<- SUCCESS! in sendReadReceipt.")
+
+    case let .failure(error):
+        print("!! FAIL! in sendReadReceipt.")
+        fatalError(error.message)
+    }
+}
+```
+
+### Get read receipts
+
+Use the `listChatReadReceipts` method to retrieve read receipts for a thread.
+
+- `chatThreadId` is the unique ID of the thread.
+
+`PagedCollection<ReadReceipt>` is the response returned from listing read receipts.
+
+```swift
+client.listChatReadReceipts(chatThreadId: threadId) { result, _ in
+    switch result {
+    case let .success(readReceipts):
+        print("<- SUCCESS! in listReadReceipts.")
+
+        var iterator = readReceipts.syncIterator
+        while let readReceipt = iterator.next() {
+            print("\tchatMessageId content=\(readReceipt.chatMessageId ?? "Not found") ")
+            print("\tsenderId content=\(readReceipt.senderId ?? "Not found") ")
+        }
+
+    case let .failure(error):
+        print("!! FAIL! in listReadReceipts.")
+        fatalError(error.message)
+    }
+}
+```
+
+# Troubleshooting
+
+Running into issues? This section should contain details as to what to do there.
```

Comment:
We can add something like what [Python](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/keyvault/azure-keyvault-certificates#troubleshooting), [Java](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/search/azure-search-documents#troubleshooting), [C#](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/storage/Azure.Storage.Blobs#troubleshooting) or [Typescript](https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/servicebus/service-bus#troubleshooting) have.

In general, they mention that clients raise certain types of exceptions and also how to log these errors.